rest/route/host_agent.go

```diff
@@ -663,6 +683,15 @@ func assignNextAvailableTask(ctx context.Context, env evergreen.Environment, tas
 			continue
 		}
 
+		err = task.SetNumNextTaskDispatches(nextTask.Id, nextTask.Execution, nextTask.NumNextTaskDispatches+1)
```

Comment:
A similar thing to the above comment, but in the case where [the host is already assigned a running task](https://github.com/evergreen-ci/evergreen/blob/8fe05b3b8f9e14487101ac82e3f308e959b8b2d7/rest/route/host_agent.go#L182-L184), we should increment the dispatch count.